ui/app/src/components/Header.tsx

```diff
@@ -11,17 +11,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { AppBar, Switch, Toolbar, Typography } from '@material-ui/core';
+import {
+  AppBar,
+  Box,
+  Button,
+  CircularProgress,
+  Divider,
+  Menu,
+  MenuItem,
+  Switch,
+  Theme,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+import { ChevronDown } from 'mdi-material-ui';
+import { SxProps } from '@mui/system/styleFunctionSx/styleFunctionSx';
+import { MouseEvent, useState } from 'react';
+import { ProjectModel, useProjectQuery } from '../model/project-client';
+import Toast from './Toast';
+
+function ProjectMenu(props: { projectList: ProjectModel[] }): JSX.Element {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleMenu = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleCloseMenu = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <>
+      <Button
+        aria-label="list of the available project"
+        aria-controls="menu-project-list-appbar"
+        aria-haspopup="true"
+        color="inherit"
+        endIcon={<ChevronDown />}
+        onClick={handleMenu}
+      >
+        Projects
+      </Button>
+      <Menu
+        id="menu-project-list-appbar"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleCloseMenu}
+      >
+        {props.projectList.map((value, index) => {
+          return (
+            // TODO when routing is in place, use the button to redirect to the project page
+            <MenuItem key={index} onClick={handleCloseMenu}>
+              {value.metadata.name}
+            </MenuItem>
+          );
+        })}
+      </Menu>
+    </>
+  );
+}
+
+const style: SxProps<Theme> = {
+  display: 'flex',
+  flexDirection: 'row',
+};
 
 export default function Header(): JSX.Element {
+  const { response, loading, error } = useProjectQuery();
   return (
     <AppBar position="relative">
       <Toolbar>
-        <Typography variant="h6" sx={{ flexGrow: 1 }}>
-          Perses
-        </Typography>
+        <Box sx={style} flexGrow={1}>
+          <Typography variant="h6" sx={{ marginRight: '1rem' }}>
+            Perses
+          </Typography>
+          <Divider
+            orientation="vertical"
+            flexItem
+            sx={{ borderRightColor: 'rgba(255,255,255,0.2)' }}
+          />
+          {loading ? (
+            <CircularProgress size="1rem" />
+          ) : (
+            response && <ProjectMenu projectList={response} />
+          )}
+        </Box>
         <Switch />
       </Toolbar>
+      {/* TODO manage in a single place the pull of fetching error */}
+      <Toast loading={loading} severity={'error'} message={error?.message} />
```

Comment:
```suggestion
      <Toast loading={loading} severity="error" message={error?.message} />
```

I see there's no `{error && ...}` here, does the toast disappear automatically if the `message` is undefined?

Reply:
It's actually in the `Toast` component that the check occurred. So if the `message` is undefined, then it won't show up.
Here: https://github.com/perses/perses/blob/main/ui/app/src/components/Toast.tsx#L26 you have the check

Reply:
I kind of a big fan of toast. I find it fit nicely to show fetch error. Probably we could add a message that is wrapping the error message. So you can identify where does it come from

Reply:
For the top menu I don't mind too much where the error is displayed, but when it comes to showing errors of individual PromQL errors, it would be nice to show them close to where the query is defined, because otherwise it's going to be harder to find out which one's failing. (I also prefer the inline query error in Prometheus over the one in Grafana, where you have to hover over a tiny area at the top left of a graph pane to see what the actual error is.)

Reply:
yeah in the case of the promql query, it would make sense + we will have the place to display the error properly.
In the Top bar it would be weird / ugly to show an error don't you think ? 

I think we can say, we use the Toast to display the fetch error when you are querying the Perses backend server.
When you are targeting the datasource, probably you want to display it in a different way for better understanding.

By the way, the toast can be used to display some success message. Like `datasource has been successfully reached`

Reply:
Yep, that sounds good to me at least.